recaf-core/src/main/java/me/coley/recaf/workspace/resource/source/ArchiveFileContentSource.java

```diff
@@ -1,157 +1,91 @@
+			entryHandler.accept(fileHeader, new LocalFileHeaderSource(fileHeader));
 		}
 	}
 
 	@Override
-	protected boolean isClass(ZipEntry entry, byte[] content) {
-		// If the entry name does not have the "CAFEBABE" magic header, its not a class.
-		return matchesClassMagic(content);
+	protected Stream<ByteSourceElement<LocalFileHeader>> stream() throws IOException {
+		ZipArchive archive = ZipIO.readJvm(getPath());
+		return archive.getLocalFiles().stream()
+				.map(x -> new ByteSourceElement<>(x, new LocalFileHeaderSource(x)));
 	}
 
 	@Override
-	protected String getPathName(ZipEntry entry) {
-		return entry.getName();
+	protected boolean isClass(LocalFileHeader fileHeader, ByteSource content) throws IOException {
+		// If the fileHeader name does not have the "CAFEBABE" magic header, its not a class.
+		return matchesClassMagic(content.peek(17));
 	}
 
-	private void handle(Path path, BiConsumer<ZipEntry, byte[]> entryHandler, boolean checkHeader) throws IOException {
-		// TODO: Use PatchingZipWriterStrategy - something like...
-		//             ZipArchive archive = ZipIO.readJvm(Files.readAllBytes(path));
-		//             ByteArrayOutputStream baos = new ByteArrayOutputStream();
-		//             new PatchingZipWriterStrategy().write(archive, baos);
-		//             byte[] fixed = baos.toByteArray();
-		//       But first we need to ensure the patcher is stable and doesn't break anything.
-		try (InputStream stream = new FileInputStream(path.toFile())) {
-			readFrom(stream, entryHandler);
-		} catch (Exception ex) {
-			logger.debug("Malformed Zip, attempting to patch: {} - {}", path, ex.getMessage());
-			checkInvalidEntryData(path, ex, entryHandler);
-			if (checkHeader)
-				checkBogusHeaderPK(path, ex, entryHandler);
-		}
+	@Override
+	protected boolean isClass(LocalFileHeader entry, byte[] content) throws IOException {
+		// If the fileHeader name does not have the "CAFEBABE" magic header, its not a class.
+		return matchesClassMagic(content);
 	}
 
-	private void checkInvalidEntryData(Path path, Exception ex, BiConsumer<ZipEntry, byte[]> entryHandler) throws IOException {
-		// Check if ZIP entries have falsified CRC values. This will cause:
-		// java.util.zip.ZipException: invalid entry CRC (expected 0xaaaaaaaa but got 0xbbbbbbbb)
-		//   at java.util.zip.ZipInputStream.readEnd(ZipInputStream.java:394)
-		//   at java.util.zip.ZipInputStream.read(ZipInputStream.java:196)
-		//   at java.io.FilterInputStream.read(FilterInputStream.java:107)
-		// For some reason using "ZipFile"/"JarFile" ignores CRC validity.
-		// Similarly, it also ignores invalid entry sizes.
-		String message = ex.getMessage();
-		if (message != null && (message.contains("invalid entry CRC") || message.contains("invalid entry size"))) {
-			try (ZipFile zf = new JarFile(path.toFile())) {
-				readFromAlt(zf, entryHandler);
-			}
-		}
+	@Override
+	protected String getPathName(LocalFileHeader entry) {
+		return entry.getFileNameAsString();
```

Comment:
We should be using the `CentralDirectoryFileHeader` since that is what the JVM trusts.

The LL-Java-Zip trick jars abuse this by putting bogus names in the `LocalFileHeader` name.